benchmarks/perf-tool/okpt/test/steps/steps.py

```diff
@@ -520,16 +698,20 @@ def recall_at_r(results, neighbor_dataset, r, k, query_count):
         Recall at R
     """
     correct = 0.0
+    total_num_of_results = 0
     for query in range(query_count):
         true_neighbors = neighbor_dataset.read(1)
         if true_neighbors is None:
             break
         true_neighbors_set = set(true_neighbors[0][:k])
-        for j in range(r):
+        true_neighbors_set.discard(-1)
+        min_r = min(r, len(true_neighbors_set))
```

Comment:
question: Why are we taking min here for recall?

Reply:
after filter got applied we decrease the number of true neighbors, and this affects the recall. Say with normal dataset we do have 5 neighbors and 2 queries, so recall should be based on 10 number of true neighbors, e.g.  
[2, 5, 6, 1, 3]
[4, 2 , 1, 7, 6]
after filter got applied with filter out some of docs and neighbors are:
[2, 6, 1]
[4, 2, 1, 6]
so we should base recall calculation on total of 7 docs, not 10.